local/tf/examples_io.py

```diff
@@ -243,7 +244,11 @@ def __load_data(self):
             idx = int(name[:-4].split('_')[1])
             label = self._train_labels[idx]
             start_time = time.time()
-            mat = np.load(self._tar.extractfile(name))
+            array_file = BytesIO()
+            array_file.write(self._tar.extractfile(name).read())
+            array_file.seek(0)
+            mat = np.load(array_file)
```

Comment:
I think you should close the stream once you've finished working with it. Otherwise, it will live in RAM forever. For convenience you can use the `BytesIO` object inside [context manager](https://docs.python.org/3/library/io.html)